### PR TITLE
RangeInfo: Rework the criterion for valid selection of multiple statements, i.e. they should have a common brace statement as parent.

### PR DESCRIPTION
**Pull request by nkcsgexi**
Suggested by @jrose-apple .

### CODE REVIEW

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

File: lib/IDE/SwiftSourceDocInfo.cpp [hunk 1]
```diff
@@ -320,6 +320,11 @@ static bool hasUnhandledError(ArrayRef<ASTNode> Nodes) {
   });
 }
 
+static bool
+hasNodeThat(ArrayRef<ASTNode> Nodes, llvm::function_ref<bool(ASTNode)> Pred) {
```
Comment (jrose-apple) [inline lib/IDE/SwiftSourceDocInfo.cpp]: This is `any_of` in STLExtras.

File: lib/IDE/SwiftSourceDocInfo.cpp [hunk 2]
```diff
@@ -342,6 +347,22 @@ struct RangeResolver::Implementation {
...
     ContextInfo(ASTNode Parent, bool ContainedInRange) : Parent(Parent),
       ContainedInRange(ContainedInRange) {}
+
+    bool isMultiStatment() {
+      if (StartMatches.empty() || EndMatches.empty())
+        return false;
+
+      // Multi-statement should have a common parent of brace statement, this
+      // can be implicit brace statement, e.g. in case statement.
+      if (Parent.isStmt(StmtKind::Brace))
```
Comment (jrose-apple) [inline lib/IDE/SwiftSourceDocInfo.cpp]: Where's the part that checks if they're all the same parent?

Comment (nkcsgexi) [inline lib/IDE/SwiftSourceDocInfo.cpp]: The is implied in `ContextInfo`; they won't end up here if they are not siblings.

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

File: lib/IDE/SwiftSourceDocInfo.cpp (hunk 1 above)
Comment (jrose-apple) [inline lib/IDE/SwiftSourceDocInfo.cpp]: No, I mean `llvm::any_of` is just your `hasNodeThat` with generics.

Comment (nkcsgexi) [inline lib/IDE/SwiftSourceDocInfo.cpp]: If that's the case, I still prefer it extracted to avoid typing `begin()` and `end()`.

Comment (jrose-apple) [inline lib/IDE/SwiftSourceDocInfo.cpp]: `llvm::any_of(StartMatches, IsCase)`. It's the same thing you have now.

**Pull request merged**